xonsh/history.py

```diff
@@ -274,38 +312,35 @@ def flush(self, at_exit=False):
         """
         if len(self.buffer) == 0:
             return
-        hf = HistoryFlusher(self.filename, tuple(self.buffer), self._queue, self._cond,
-                            at_exit=at_exit)
+        hf = HistoryFlusher(self.filename, tuple(self.buffer), self._queue,
+                            self._cond, at_exit=at_exit)
         self.buffer.clear()
         return hf
 
+
 #
 # Interface to History
 #
-
-_HIST_PARSER = None
```

Comment:
I guess you could get the same behaviour with `functools.lru_cache()`, but still...